docs/source/transaction_family_specifications/track_and_trade_transaction_family.rst

```diff
@@ -0,0 +1,717 @@
+************************************************
+Track and Trade Transaction Family Specification
+************************************************
+
+Overview
+========
+
+The Track and Trade (T&T) transaction family allows users to track
+goods as they move through a supply chain. Records for goods include a
+history of ownership and custodianship, as well as histories for a
+variety of properties such a temperature and location. These
+properties are managed through a user-specifiable system of record
+types.
+
+
+State
+=====
+
+All T&T objects are serialized using Protocol Buffers before being
+stored in state. These objects include: Agents, Properties
+(accompanied by their auxiliary PropertyPage objects), Proposals,
+Records, and RecordTypes. As described in the Addressing section
+below, these objects are stored in separate sub-namespaces under the
+T&T namespace. To handle hash collisions, all objects are stored in
+lists within protobuf "Container" objects, the order of which is
+described below.
+
+
+Records
+-------
+
+Records represent the goods being tracked by Track and Trade. Almost
+every transaction references some Record.
+
+A Record contains a unique identifier, the name of a RecordType, and
+lists containing the history of its owners and custodians. It also
+contains a `final` flag indicating whether further updates can be made
+to the Record and its Properties. If flag is set to true, then no
+further updates can be made to the Record, including changing its
+`final` flag.
+
+.. code-block:: protobuf
+
+   message Record {
+       message AssociatedAgent {
+           // The Agent's public key
+           string agent_id = 1;
+
+           // Approximately when this agent was associated, as a Unix UTC timestamp
+           uint64 timestamp = 2;
+       }
+
+       // The unique user-defined natural key which identifies the
+       // object in the real world (for example, a serial number)
+       string identifier = 1;
+
+       string record_type = 2;
+
+       // Ordered oldest to newest by timestamp
+       repeated AssociatedAgent owners = 3;
+       repeated AssociatedAgent custodians = 4;
+
+       // Flag indicating whether the Record can be updated. If it is set
+       // to true, then the record has been finalized and no further
+       // changes can be made to it or its Properties.
+       bool final = 5;
+   }
+
+
+Note that while information about a Record's owners and custodians are
+included in the object, information about its Properties are stored
+separately (see below).
+
+Records whose addresses collide will be stored in a list
```

Comment:
Should "will be" be changed to "are"?  I usually expect to see the present tense in a spec (as opposed to the future tense in a design document).  This comment applies to the other occurrences of "will be" below, though some should be "is".

Reply:
This gets into a philosophical issue: is the purpose of the spec to describe behavior, to prescribe behavior, or to define behavior? Some of the other specs use the word `shall`, which I think is a little overblown, but I take it that the sense is prescriptive. Another prescriptive wording would be something like `Records whose addresses collide are to be stored...`. A definitional approach might be `With a spec-compliant TP, records whose addresses collide are stored...`.

All that said, I don't have any particularly strong opinions. I'll do it whichever way you prefer, unless someone else objects, in which case I'll do it according to the opinion of the winner of the resulting fight to the death.

Reply:
This is completely unrelated to software, but for a similar grammatical issue, consider this line from the Declaration of Independence: `[A]ll political connection between [the Colonies] and the State of Great Britain, is and ought to be totally dissolved`.

In that sentence, what is the sense of the word `is`?

Reply:
Tech writers have strong opinions about using present tense (and fewer words) whenever possible.  (They train us in minimalist writing, but those courses don't include battle skills.). If using "is" or "are" is clear, I'd vote for that.

Reply:
As far as the Declaration of Independence, I'm pleased that they didn't say "shall", because that could be interpreted as the future continuous tense.  Using "is" is direct and declarative; it means "now". The connection is dissolved as of the declaration. 

I love discussing verb tense and mood in pull-request comments. But I'll stop now.